src/threading/MsgThread.cc

```diff
@@ -341,46 +343,122 @@ void MsgThread::Finished()
 
 void MsgThread::Info(const char* msg)
 	{
-	SendOut(new detail::ReporterMessage(detail::ReporterMessage::INFO, this, msg));
+	ODesc desc;
+
+	if ( auto* location = GetLocationInfo() )
+		{
+		location->Describe(&desc);
+		desc.Add(": ");
+		}
+
+	desc.Add(msg);
+	SendOut(new detail::ReporterMessage(detail::ReporterMessage::INFO, this, desc.Description()));
 	}
 
 void MsgThread::Warning(const char* msg)
 	{
-	SendOut(new detail::ReporterMessage(detail::ReporterMessage::WARNING, this, msg));
+	ODesc desc;
+
+	if ( auto* location = GetLocationInfo() )
+		{
+		location->Describe(&desc);
+		desc.Add(": ");
+		}
+
+	desc.Add(msg);
```

Comment:
This code is repeated a  number  of times across these methods, suggest to factor it out.